src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py

```diff
@@ -40,32 +40,61 @@ def _update_config_params(self, config):
         build["subsampling_scheme"] = self.subsampling_scheme
 
         # Update the tree's title with build type, location and date range.
-        if (self.template_args.get("filter_start_date") is not None) and (
-            self.template_args.get("filter_end_date") is not None
-        ):
+        # We always provide some form of end date in the title.
+        end_date = self._get_formatted_tree_end_date()
+        # We base format of title on whether we have a `filter_start_date`
+        if self.template_args.get("filter_start_date") is not None:
             title_template = "{tree_type} tree for samples collected in {location} between {start_date} and {end_date}"
             build["title"] = title_template.format(
                 tree_type=self.subsampling_scheme.title(),
                 location=", ".join(location_values),
                 start_date=dateparser.parse(
                     self.template_args.get("filter_start_date")
                 ).strftime("%Y-%m-%d"),
-                end_date=dateparser.parse(
-                    self.template_args.get("filter_end_date")
-                ).strftime("%Y-%m-%d"),
+                end_date=end_date,
             )
         else:
-            title_template = "{tree_type} tree for samples collected in {location}"
+            title_template = "{tree_type} tree for samples collected in {location} up until {end_date}"
             build["title"] = title_template.format(
                 tree_type=self.subsampling_scheme.title(),
                 location=", ".join(location_values),
+                end_date=end_date,
             )
 
         config["files"]["description"] = config["files"]["description"].format(
             tree_type=self.subsampling_scheme.lower()
         )
         config["priorities"]["crowding_penalty"] = self.crowding_penalty
 
+    def _get_formatted_tree_end_date(self):
```

Comment:
Out of curiosity, when do you guys use "_" in the beginning of the method name? I see this in nextstrain repo a lot

Reply:
Sorry for the delay: it's basically a convention in Python to say "hey, this is really just intended for internal use, don't touch this thing unless you really know what you're doing". In other languages, there's the concept of "Public" and "Private" methods -- you can import/access "Public" stuff elsewhere, but "Private" is not directly accessible -- but Python does not have that. So it's a stylistic convention that most people in the Python space are aware of and follow (you'll also see it in other languages, I'm pretty sure I've seen it in the wild in JavaScript code, etc).